packages/hotpotqa/src/aviary/envs/hotpotqa/env.py

```diff
@@ -309,20 +308,17 @@ async def step(
 
         # We accumulate reward across all tool calls in a step
         self.state.reward = 0.0
-        valid_action, invalid_action = self.filter_invalid_tool_calls(action)
-        # NOTE: valid_action or invalid_action may have an empty list of tool calls
         response_messages = cast(
             Messages,
             # Ordered since things like search -> lookup need to be run in order.
             # NOTE: Handling tool exceptions here keeps the trajectory going, but I don't
             # think the returned message is useful to the agent/learning. Disabling for now.
             await self.exec_tool_calls(
-                valid_action, ordered=True, handle_tool_exc=False
-            )
-            + [
-                ToolResponseMessage.from_call(tool_call, content="Invalid tool call.")
-                for tool_call in invalid_action.tool_calls
-            ],
+                action,
+                ordered=True,
+                handle_tool_exc=False,
+                handle_invalid_tool_calls=True,
```

Comment:
Can you do GSM8k too?

Reply:
Uh - I looked at gsm8k code. It's insane - why do we `json.loads(result.content)`? I do not understand what is happening in the code and thus cannot modify it during this PR. Will need to understand better what is going on

Reply:
That's fine. Then since `handle_invalid_tool_calls` is `True` by default, can you opt out of this in GSM8k? Or do you think it's fine to opt into

Reply:
Sure!

Reply:
Yeah I'll tackle this - this is all a holdover from when I first started working with Aviary and did some anti-idiomatic things. Will fix.